Modules/Common/SolanaTraits.php

```diff
@@ -0,0 +1,364 @@
+<?php declare(strict_types = 1);
+
+/*  Idea (c) 2023 Nikita Zhavoronkov, nikzh@nikzh.com
+ *  Copyright (c) 2023-2024 3xpl developers, 3@3xpl.com, see CONTRIBUTORS.md
+ *  Distributed under the MIT software license, see LICENSE.md  */
+
+/*  Common Solana functions and enums  */
+
+require_once __DIR__ . '/../../Engine/Crypto/Base58.php';
+require_once __DIR__ . '/../../Engine/Crypto/Sodium.php';
+
+Enum SolanaAddressPrograms: string
```

Comment:
Not sure if this belongs to a top-level class. See Beacon Chain modules for example: chain-specific settings are on the final class level. We can have a separate trait just for Solana though, but not for Solana-like.

Reply:
Let's keeep it there, as it is common for all solana modules, even if we are not using it in the main module, it is actual for it also, and I will need to duplicate it in 3 classes, also this enum is used in traits functions.